package.json

```diff
@@ -110,10 +110,11 @@
   "scripts": {
     "start": "cross-env PORT=3001 craco start",
     "start:test": "cross-env PORT=3001 REACT_APP_TEST_MODE=true craco start",
-    "start:for-real-backend": "cross-env REACT_APP_BACKEND=http://localhost:3000 yarn start",
+    "start:for-real-backend": "cross-env PORT=3001 REACT_APP_BACKEND_BASE_URL=http://localhost:3000 craco start",
     "serve:build": "http-server build/ -s -p 3001 -P \"http://localhost:3001?\"",
-    "build:test": "cross-env REACT_APP_TEST_MODE=true craco build",
-    "build:production": "craco build",
+    "build:test": "cross-env REACT_APP_BACKEND_BASE_URL=/mock-backend REACT_APP_TEST_MODE=true craco build",
+    "build:mock": "cross-env REACT_APP_BACKEND_BASE_URL=/mock-backend craco build",
+    "build:production": "[[ -v REACT_APP_BACKEND_BASE_URL ]] && (cross-env craco build) || (echo -e '\n==============\nREACT_APP_BACKEND_BASE_URL not set.\nUse this task only if you want to create a production build with a real backend. Otherwise use build:mock\n==============\n'; exit 1)",
```

Comment:
(npm scripts are no bash scripts but shell scripts)

Reply:
![image](https://user-images.githubusercontent.com/6124140/116795313-57eef980-aad4-11eb-9657-48213b38bebf.png)
Nope. it works

Reply:
You're running on macOS, right?

Reply:
I added a force for bash

Reply:
My suggested change would be to remove the bash specifics as well as do proper escaping:

```diff
diff --git a/package.json b/package.json
--- a/package.json	(revision e2dc45ad89f9668670eea91bf105dce838a6d273)
+++ b/package.json	(date 1619904506990)
@@ -114,7 +114,7 @@
     "serve:build": "http-server build/ -s -p 3001 -P \"http://localhost:3001?\"",
     "build:test": "cross-env REACT_APP_BACKEND_BASE_URL=/mock-backend REACT_APP_TEST_MODE=true craco build",
     "build:mock": "cross-env REACT_APP_BACKEND_BASE_URL=/mock-backend craco build",
-    "build:production": "[[ -v REACT_APP_BACKEND_BASE_URL ]] && (cross-env craco build) || (echo -e '\n==============\nREACT_APP_BACKEND_BASE_URL not set.\nUse this task only if you want to create a production build with a real backend. Otherwise use build:mock\n==============\n'; exit 1)",
+    "build:production": "[ -z \"$REACT_APP_BACKEND_BASE_URL\" ] && (cross-env craco build) || (echo -e '\\n==============\\nREACT_APP_BACKEND_BASE_URL not set.\\nUse this task only if you want to create a production build with a real backend. Otherwise use build:mock\\n==============\\n'; exit 1)",
     "analyze": "cross-env ANALYZE=true craco build",
     "test": "craco test",
     "lint": "eslint --max-warnings=0 --ext .ts,.tsx src",

```